src/flux/course-listing.coffee

```diff
@@ -70,4 +70,9 @@ CourseListingStore = flux.createStore
     allCourses: ->
       return _.compact _.map @_course_ids, CourseStore.get
 
+    allCoursesWithRoles: ->
+      return _.compact _.map @_course_ids, (id) ->
+        course = CourseStore.get(id)
+        course if course?.roles?.length
+
```

Comment:
although I suppose our lint rules say it should be `not _.isEmpty( CourseStore.get(id)?.roles )` 😀